lib/active_merchant/billing/gateways/spreedly_core.rb

```diff
@@ -155,12 +143,12 @@ def save_card(retain, credit_card, options)
       end
 
       def purchase_with_token(money, payment_method_token, options)
-        request = auth_purchase_request(money, payment_method_token, options)
+        request = build_purchase_request(money, payment_method_token, options)
```

Comment:
@lancecarlson I'd prefer that we retain `auth` in the method name since it is dual purpose. I'm fine with `build_auth_purchase_request`.

Reply:
@jknipp What if we consolidated things a bit and made a method like this:

```
      def build_request(money, payment_method, options)
        build_xml_request('transaction') do |doc|
          add_invoice(doc, money, options)
          add_payment_method(doc, payment_method, options)
          retain_on_success(doc, true) if options[:store]
          add_extra_options(:gateway_specific_fields, doc, options)
        end
      end
```

Then we can chop down ```verify_with_token```. If you're cool with this refactoring, I will start getting the tests to pass. :)

Reply:
👍 Sound good to me, I like the idea of dry'ing up the code here.